### PR TITLE
[Doc] Update QuickStart

### PR DESCRIPTION
**Pull request by EsoragotoSpirit**
Signed-off-by: EsoragotoSpirit <wanglichen@starrocks.com>

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: docs/quick_start/Deploy.md [hunk 1]
```diff
@@ -15,17 +15,22 @@ Following these steps, you can deploy a StarRocks instance with only one fronten
 Before deploying StarRocks, make sure the following requirements are satisfied.
 
 - **Hardware**
-  You can follow these steps on relatively elementary hardware, such as a machine with 4 CPU cores and 16 GB of RAM. The CPU MUST support AVX2 instruction sets.
+
+  You can follow these steps on relatively elementary hardware, such as a machine with 4 CPU cores and 16 GB of RAM. In production, we recommend you run StarRocks with CPUs that support the AVX2 instruction sets. Due to a lack of support for the AVX2 instruction sets, ARM architecture is less competitive than x86 architecture in some scenarios.
```
Comment (ss892714028) [inline docs/quick_start/Deploy.md:19]: The binary from SR's website does not support arm as well

File: docs/quick_start/Deploy.md [hunk 2]
```diff
@@ -15,17 +15,22 @@ Following these steps, you can deploy a StarRocks instance with only one fronten
...
 
 > **NOTE**
 >
 > You can run `cat /proc/cpuinfo | grep avx2` in your terminal to check if the CPU supports the AVX2 instruction sets.
 
 - **Software**
-  Your machine MUST be running on OS with Linux kernel 3.10 or later. In addition, you must have JDK 1.8 or later and MySQL client 5.5 or later installed on your machine. Please note that JRE is not supported.
+
+  Your machine MUST be running on CentOS 7.0 or later. To run StarRocks on other operating systems, you must [Deploy StarRocks in Docker](../quick_start/deploy_in_docker.md).
```
Comment (ss892714028) [inline docs/quick_start/Deploy.md:27]: To run on Ubuntu 22.04 or with ARM-based processors, from [here](https://hub.docker.com/r/starrocks/allin1-ubuntu), find the tag, run the image, and copy the binaries from the docker container to your host machine.

Comment (ss892714028) [reply docs/quick_start/Deploy.md:27]: FE dir: /data/starrocks/fe/
BE dir: /data/starrocks/be/

**Review by ss892714028 — CHANGES_REQUESTED** [review]

**Review by evelynzhaojie — APPROVED** [review]

**Review by amber-create — APPROVED** [review]

**Review by ss892714028 — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**